tests/test_templates_layers.py

```diff
@@ -462,3 +462,74 @@ def test_random_layer_weights(self, n_subsystems, tol):
         params = [q.parameters for q in rec.queue]
         params_flat = [item for p in params for item in p]
         assert np.allclose(weights.flatten(), params_flat, atol=tol)
+
+
+class TestCnotRing:
+    """Tests for the BasicEntanglerLayers method from the pennylane.templates.layers module."""
+
+    @pytest.mark.parametrize("n_wires, n_cnots", [(1, 0),
+                                                  (2, 1),
+                                                  (3, 3),
+                                                  (4, 4)])
+    def test_circuit_queue(self, n_wires, n_cnots):
```

Comment:
Separating this into two tests: `test_circuit_queue_types` and `test_circuit_queue_parameter_values`, could make any errors arising be more explicit.